language/move-vm/types/src/values/values_impl.rs

```diff
@@ -847,6 +847,13 @@ impl Value {
     pub fn struct_(s: Struct) -> Self {
         Self(ValueImpl::new_container(s.0))
     }
+
+    // TODO: This is a temporary API to allow vector<u8> to be created externally.
```

Comment:
It's on the next two lines: I wonder if we should instead have a `fn vector(Vec<Values>)` instead of specialized constructors like `fn vector_u8(Vec<u8>)`. The problem with `Vec<Values>` is that we need to check that all input values have the same layout, or otherwise we're breaking some of the guarantees the vm-runtime-types offer.

Hmm... now on a second glance I actually think it's a good idea to have a `vector_u8` -- simpler, safer and easier to use than the general version. Let me change the comment to "TODO: consider if we want to replace this with with `fn vector(Vec<Value>)`".